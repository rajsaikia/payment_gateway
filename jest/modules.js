// https://react-hook-form.com/faqs#TestingfailedduetoMutationObserver
import 'mutationobserver-shim';

// Expanded expect library required for testing-library
import '@testing-library/jest-dom/extend-expect';
