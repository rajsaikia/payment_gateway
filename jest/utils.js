import React from 'react';
import PropTypes from 'prop-types';
import { render as standardRender } from '@testing-library/react';
import { loadStripe } from '@stripe/stripe-js';
import { Elements } from '@stripe/react-stripe-js';

const ELEMENTS_OPTIONS = {
	fonts: [
		{
			cssSrc: 'https://fonts.googleapis.com/css?family=Roboto',
		},
	],
};

const stripePromise = loadStripe(process.env.REACT_APP_SECRAT_KEY);

export const AllProviders = ({ children }) => (
	<Elements stripe={stripePromise} options={ELEMENTS_OPTIONS}>
		<>{children}</>
	</Elements>
);

AllProviders.propTypes = {
	children: PropTypes.any,
};

const customRender = (ui, options) =>
	standardRender(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
