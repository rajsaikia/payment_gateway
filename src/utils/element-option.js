const ELEMENTS_OPTIONS = {
	fonts: [
		{
			cssSrc: 'https://fonts.googleapis.com/css?family=Roboto',
		},
	],
};

export default ELEMENTS_OPTIONS;
