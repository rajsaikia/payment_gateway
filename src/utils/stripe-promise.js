import { loadStripe } from '@stripe/stripe-js';
const StripePromise = loadStripe(process.env.REACT_APP_SECRAT_KEY);

export default StripePromise;
