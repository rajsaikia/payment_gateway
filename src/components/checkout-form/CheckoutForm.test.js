import React from 'react';
import { render } from 'jest/utils';
import '@testing-library/jest-dom/extend-expect';
import CheckoutForm from './index';

describe('<CheckoutForm /> - Component', () => {
	let container = null;

	const Component = () => {
		return <CheckoutForm />;
	};
	beforeEach(async () => {
		container = render(<Component />);
	});

	afterEach(() => {
		container.unmount();
	});
	it('renders correctly', async () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
	it('it should contain the text Name', async () => {
		expect(container.getByText('Name')).toBeInTheDocument();
	});
	it('it should contain the text Email', async () => {
		expect(container.getByText('Email')).toBeInTheDocument();
	});
	it('it should contain the text Phone', async () => {
		expect(container.getByText('Phone')).toBeInTheDocument();
	});
	it('it should contain the text Pay $25', async () => {
		expect(container.getByText('Pay $25')).toBeInTheDocument();
	});
});
