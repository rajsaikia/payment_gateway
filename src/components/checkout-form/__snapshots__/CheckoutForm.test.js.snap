// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutForm /> - Component renders correctly 1`] = `
<DocumentFragment>
  <form
    class="Form"
  >
    <fieldset
      class="FormGroup"
    >
      <div
        class="FormRow"
      >
        <label
          class="FormRowLabel"
          for="name"
        >
          Name
        </label>
        <input
          autocomplete="name"
          class="FormRowInput"
          id="name"
          placeholder="Jane Doe"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="FormRow"
      >
        <label
          class="FormRowLabel"
          for="email"
        >
          Email
        </label>
        <input
          autocomplete="email"
          class="FormRowInput"
          id="email"
          placeholder="janedoe@gmail.com"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="FormRow"
      >
        <label
          class="FormRowLabel"
          for="phone"
        >
          Phone
        </label>
        <input
          autocomplete="tel"
          class="FormRowInput"
          id="phone"
          placeholder="(941) 555-0123"
          required=""
          type="tel"
          value=""
        />
      </div>
    </fieldset>
    <fieldset
      class="FormGroup"
    >
      <div
        class="FormRow"
      >
        <div />
      </div>
    </fieldset>
    <button
      class="SubmitButton "
      disabled=""
      type="submit"
    >
      Pay $25
    </button>
  </form>
</DocumentFragment>
`;
